src/include/processor/operator/persistent/reader/csv/csv_error.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "common/uniq_lock.h"
+#include "processor/warning_context.h"
+
+namespace kuzu::processor {
+
+class BaseCSVReader;
+
+struct LineContext {
+    uint64_t startByteOffset;
+    uint64_t endByteOffset;
+
+    bool isCompleteLine;
+
+    void setNewLine(uint64_t start);
+    void setEndOfLine(uint64_t end);
+};
+
+// If parsing in parallel during parsing we may not be able to determine line numbers
+// Thus we have additional fields that can be used to determine line numbers + reconstruct lines
+// After parsing this will be used to populate a PopulatedCSVError instance
+struct CSVError {
+    std::string message;
+    std::string filePath;
+    LineContext errorLine;
+
+    uint64_t blockIdx;
+    uint64_t numRowsReadInBlock;
+
+    bool operator<(const CSVError& o) const;
+};
+
+using warning_counter_t = uint64_t;
+
+struct LinesPerBlock {
+    uint64_t validLines;
+    uint64_t invalidLines;
+    bool doneParsingBlock;
+};
+
+class CSVErrorHandler {
```

Comment:
Consider renaming this class to `CSVFileErrorHandler` since it's handling errors of a file.

Reply:
updated